packages/core/src/provider.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { createContext, useCallback, useContext, useEffect, useMemo, useReducer } from 'react';
+import type { ReactNode } from 'react';
+import { AElfAddress, AElfDappBridge, PublicKey } from '@aelf-react/types';
+import { isMobileDevices } from './utils/isMobile';
+import { formatLoginInfo } from './utils';
+
+const ACTIVATE = 'activate';
+const DEACTIVATE = 'deactivate';
+
+const Listeners = ['nightElfRemoveKeyPairs', 'nightElfLockWallet'];
+
+/**
+ * @param children - A React subtree that needs access to the context.
+ * @param appName - App name.
+ * @param nodes - node object. @example `nodes = {AELF: {rpcUrl:'xxx', chainId:"AELF"}, tDVV: {rpcUrl:'xxx', chainId:"tDVV"}}`
+ */
+export interface AElfReactProviderProps {
+  children: ReactNode;
+  appName: string;
+  nodes?: {
+    [key: string]: {
+      rpcUrl: string;
+      chainId: string;
+    };
+  };
+}
+
+export interface AElfContextType {
+  name?: string;
+  chainId?: string;
+  account?: AElfAddress;
+  defaultAElfBridge?: AElfDappBridge;
+  aelfBridges?: { [key: string]: AElfDappBridge };
+  isActive: boolean;
+  pubKey?: string;
+  publicKey?: PublicKey;
+  activate: (nodes?: AElfReactProviderProps['nodes']) => Promise<true | any>;
+  deactivate: () => Promise<true | any>;
+  nodes?: AElfReactProviderProps['nodes'];
+}
+
+const INITIAL_STATE = {
+  isActive: false,
+  account: undefined,
+  defaultAElfBridge: undefined,
+  aelfBridges: undefined,
+  pubKey: undefined,
+  publicKey: undefined,
+};
+
+const AElfContext = createContext<any>(undefined);
+
+//reducer
+function reducer(state: any, { type, payload }: any) {
+  switch (type) {
+    case DEACTIVATE: {
+      return Object.assign({}, state, INITIAL_STATE, payload);
+    }
+    default: {
+      return Object.assign({}, state, payload);
+    }
+  }
+}
+
+export function AElfReactProvider({ children, appName, nodes: providerNodes }: AElfReactProviderProps) {
+  const [state, dispatch]: [AElfContextType, any] = useReducer(reducer, INITIAL_STATE);
```

Comment:
please declare
`const {account, defaultAElfBridge, xxx,...xxx} = state` instead of `state.account || !state.defaultAElfBridge`